### PR TITLE
ci: Automate DID-resolver deployments [DEV-1263]

### PR DESCRIPTION
**Pull request by filipdjokic**
- Modify GitHub actions to push to DOCR
- Automate deployments to staging and production environments

### CODE REVIEW

File: .github/workflows/deploy-staging.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+      - name: Build and push image
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          file: docker/Dockerfile
+          platforms: linux/amd64
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
```
Comment (ankurdotb) [inline .github/workflows/deploy-staging.yml:76]: Do we really need this? We do a Docker build in the build stage. Release should automatically be pushing to Github container registry. So, at this stage, we shouldn't push to gchr.io and to be honest we shouldn't need to rebuild either since we can just push the file made in build stage to DOCR for staging deploy.

File: .github/workflows/deploy-staging.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+          
+  deploy-staging-resolver:
+    name: "Deploy DID-resolver to the staging environment"
+    needs: release-docker
+    runs-on: ubuntu-latest
+    environment:
+      name: staging
+      url: https://did-resolver-dev-39yxa.ondigitalocean.app
```
Comment (ankurdotb) [inline .github/workflows/deploy-staging.yml:84]: Just make this a simpler, custom URL. Please set it to resolver-staging.cheqd.net rather than this random DO one or the .dev one.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -86,11 +91,16 @@ jobs:
...
+        run: doctl registry login --expiry-seconds 600
+      
       - name: Configure Docker image metadata
         id: meta
         uses: docker/metadata-action@v4
         with:
-          images: ghcr.io/${{ github.repository}}
+          images: |
+            ghcr.io/${{ github.repository}}
+            registry.digitalocean.com/${{ github.repository }}
```
Comment (ankurdotb) [inline .github/workflows/release.yml:103]: Unsure if build push action will successfully publish to both GCHR and DOCR. I know that GCHR is built-in, but DOCR pushes as far as I understand can only be done using DOCTL but maybe I'm wrong.

**Review by ankurdotb — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 3 above)
Comment (filipdjokic) [reply .github/workflows/release.yml:103]: This is just for metadata action and we have a separate docker login step with doctl. 
We use the same approach on [bdjuno](https://github.com/cheqd/bdjuno/blob/chains/cheqd/testnet/.github/workflows/release.yml#L48) and it works fine for now.

I saw on docker-metadata action documentation that they support multiple images so I think this should work.

File: .github/workflows/deploy-staging.yml (hunk 2 above)
Comment (filipdjokic) [reply .github/workflows/deploy-staging.yml:84]: Agree.

File: .github/workflows/deploy-staging.yml (hunk 1 above)
Comment (ankurdotb) [reply .github/workflows/deploy-staging.yml:76]: @filipdjokic Can you take a look and resolve this?

File: .github/workflows/deploy-staging.yml (hunk 2 above)
Comment (ankurdotb) [reply .github/workflows/deploy-staging.yml:84]: Then change it?

File: .github/workflows/release.yml (hunk 3 above)
Comment (ankurdotb) [reply .github/workflows/release.yml:103]: Let me rephrase the question and actually should have tagged a different section: to publish, if you use `build-push-action`, I think that will only be able to publish to support registry like GCHR. And you need to do a doctl push separately. Correct?

File: .github/workflows/deploy-staging.yml (hunk 2 above)
Comment (filipdjokic) [reply .github/workflows/deploy-staging.yml]: I did it here [https://github.com/cheqd/did-resolver/commit/504886411a3d6a358ab0302f666d68d9bcabe7af](url), PR is not updated yet.

File: .github/workflows/deploy-staging.yml (hunk 1 above)
Comment (filipdjokic) [reply .github/workflows/deploy-staging.yml]: I removed pushing to ghcr, there is no need to do that.

I'm not sure about removing the docker build from this phase only because of the stuff we talked about before (when developers are working on multiple branches at the same time).

This is a manually triggered workflow and it wouldn't be easy to find specific branch artifacts and it would require some kind of naming (also I'm not sure if it's possible to share artifacts between workflows).

With the current setup, developers would just enter a branch name they want to deploy, action will checkout to that branch and push with the staging-latest tag.

It will run slightly longer, but I guess it will give us more flexibility.

File: .github/workflows/release.yml (hunk 3 above)
Comment (filipdjokic) [reply .github/workflows/release.yml:103]: You can push to DOCR with regular `docker push`, you just need separate docker login with doctl.

Tbh, I didn't test it yet with `build-push-action` but if they use the same underlying engine it should work.

I will make a separate step for DOCR anyway, just in case they don't support it.

**Review by ankurdotb — APPROVED** [review]
